shell/browser/electron_navigation_throttle.cc

```diff
@@ -37,11 +39,26 @@ ElectronNavigationThrottle::WillStartRequest() {
     return PROCEED;
   }
 
-  if (handle->IsRendererInitiated() &&
+  bool is_renderer_initiated = handle->IsRendererInitiated();
+
+  // Chrome's internal pages always mark navigation as browser-initiated. Let's
+  // ignore that.
+  // https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/navigator.cc;l=883-892;drc=0c8ffbe78dc0ef2047849e45cefb3f621043e956
+  if (!is_renderer_initiated &&
+      ui::PageTransitionIsWebTriggerable(handle->GetPageTransition())) {
+    auto* render_frame_host = contents->GetPrimaryMainFrame();
+    auto* rfh_impl =
+        static_cast<content::RenderFrameHostImpl*>(render_frame_host);
+    if (rfh_impl && rfh_impl->web_ui()) {
+      is_renderer_initiated = true;
+    }
```

Comment:
_(nit)_ can be slightly simpler:

```suggestion
    is_renderer_initiated = rfh_impl && rfh_impl->web_ui();
```